README.md

```diff
@@ -12,38 +12,70 @@ It is expected that you have an environment where ArduPilot can be built. Otherw
 
 1. Fork and Clone the ArduPilot/CustomBuild repository
 
-2. In the directory containg the CustomBuild repo just cloned, create a /base subdirectory and change to that directory.
+2. In the directory containing the CustomBuild repo just cloned, create a /base subdirectory and change to that directory.
 
-3. Clone a fork of the ArduPilot/ardupilot repository. The structure should appear similar to that below:
+3. Clone a fork of the ArduPilot/ardupilot repository.
 
+### Detailed Instructions for Ubuntu
+
+For example, here are the commands for Ubuntu.
+
+If not already set up, [configure your Git username](https://git-scm.com/docs/git-config#Documentation/git-config.txt-credentialusername):
+Substitute `your-github-account` with your Github username.
+```
+git config --global credential.username your-github-account
+```
+
+Next, clone the repositories:
+```bash
+git config credential.username your-github-account
+cd ~
+git clone git@github.com:$(git config credential.username)/CustomBuild.git
+git clone --depth 1 git@github.com:$(git config credential.username)/ardupilot.git base/ardupilot
+```
+
+Finally, add the upstream ArduPilot to your remotes.
+```bash
+git -C base/ardupilot remote add upstream git@github.com:ardupilot/ardupilot.git
+git -C base/ardupilot fetch --depth 1 upstream
+```
 
 ### Directory structure
-default directory structure is as follows
+
+The default directory structure is as follows:
 ```
 /home/<username>
 -CustomBuild
 -base
 --ardupilot
 ```
-### Install Flask
-```
-python3 -m pip install --user -U flask
+
+
+### Install Dependencies
+
+```bash
+cd ~/CustomBuild
+python3 -m pip install --user --upgrade --requirement requirements.txt
```

Comment:
Isn't this going to hit venv problems on Ubuntu - breaking system package warnings and whatnot?  Not a blocker as the original instructions would suffer the same issue.

Reply:
I tested the exact commands no Ubuntu 22.04 host and it worked fine. With a venv, you just omit the `--user`, which is pretty obvious from the command line errors. 

It works and it follows the current workflow. Not perfect, but venv is out of scope of my intent to get something usable.